core/src/main/java/com/linecorp/armeria/client/ClientThreadLocalState.java

```diff
@@ -129,6 +129,13 @@ private static void reportThreadSafetyViolation() {
             this.oldCaptor = oldCaptor;
         }
 
+        void add(ClientRequestContext ctx) {
+            captured.add(ctx);
+            if (oldCaptor != null) {
+                oldCaptor.add(ctx);
+            }
+        }
```

Comment:
How about not adding after `close()`?

Reply:
Oops. 😅

Reply:
Changed. But I couldn't make a test case because `oldCaptor.add()` is never called...